### PR TITLE
redolog: alwasy set running to false when close

### PR DESCRIPTION
**Pull request by ben1009**
### What problem does this PR solve? 
close https://github.com/pingcap/ticdc/issues/3412

### What is changed and how it works?
alwasy set running to false when close

### Check List 

Tests 

 - Unit test

### Release note 

```
None
```

### CODE REVIEW

File: cdc/redo/writer/file.go [hunk 1]
```diff
@@ -255,13 +255,13 @@ func (w *Writer) Close() error {
 	}
 	w.Lock()
 	defer w.Unlock()
+	defer w.running.Store(false)
 
 	err := w.close()
 	if err != nil {
 		return err
 	}
```
Comment (amyangfei) [inline cdc/redo/writer/file.go:263]: Can we add a unit test to check `w.close` returns nil error and `running` state is set properly.

**Review by amyangfei — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Comment by ben1009** [comment]
> /run-kafka-integration-test

FAIL	github.com/pingcap/ticdc/dm/dm/master	91.599s

**Pull request merged**